src/durmod/kaldi-durmod.cc

```diff
@@ -0,0 +1,466 @@
+  feat_maker.MakeFeatureVector(phone_dur_context, phone_index, &feat);
+  int32 phone_duration = phone_dur_context[phone_index].second;
+  SparseMatrix<BaseFloat> feat_mat(1, feat.Dim());
+  feat_mat.SetRow(0, feat);
+  int32 output_dim = feat_maker.OutputDim();
+  Posterior output_elements(1);
+
+  // The nodes at the output of the network are indexed from
+  // 0 to (max_duration_ - 1). Index 0 is for duration = 1.
+  int32 duration_id = (phone_duration > output_dim) ?
+                                                     (output_dim - 1):
+                                                     (phone_duration - 1);
+  output_elements[0].push_back(std::make_pair(duration_id, 1.0));
+  SparseMatrix<BaseFloat> output_mat(output_dim, output_elements);
+
+  NnetIo input, output;
+  input.name = "input";
+  input.features = feat_mat;
+  input.indexes.resize(1);
+  output.name = "output";
+  output.features = output_mat;
+  output.indexes.resize(1);
+  eg->io.push_back(input);
+  eg->io.push_back(output);
+}
+
+void AlignmentToNnetExamples(const PhoneDurationFeatureMaker &feat_maker,
+                         const std::vector<std::pair<int32, int32> > &alignment,
+                         std::vector<NnetExample> *egs) {
+  for (int i = 0; i < alignment.size(); i++) {
+    NnetExample eg;
+    MakeNnetExample(feat_maker, alignment, i, &eg);
+    egs->push_back(eg);
+  }
+}
+
+std::pair<int32, int32> OlabelToPhoneAndDuration(fst::StdArc::Label olabel,
+                                                 int32 num_phones) {
+  return std::make_pair(olabel % (num_phones + 1),
+                        olabel / (num_phones + 1));
+}
+
+fst::StdArc::Label PhoneAndDurationToOlabel(int32 phone,
+                                            int32 duration,
+                                            int32 num_phones) {
+  return phone + duration * (num_phones + 1);
+}
+
+void DurationModelReplaceLabelsLattice(CompactLattice *clat,
+                                       const TransitionModel &tmodel,
+                                       int32 num_phones) {
+  // iterate over all arcs
+  for (int s = 0; s < clat->NumStates(); s++) {
+    for (fst::MutableArcIterator<CompactLattice> aiter(clat, s);
+         !aiter.Done(); aiter.Next()) {
+      const CompactLatticeArc& arc = aiter.Value();
+
+      // find the duration and phone-id of the phone on the arc
+      const std::vector<int32> &tid_seq = arc.weight.String();
+      int32 duration = tid_seq.size();
+      int32 phone_label = 0;
+      if (duration != 0)  // if not an epsilon arc
+        phone_label = tmodel.TransitionIdToPhone(tid_seq[0]);
+
+      // encode phone+duration to 1 integer as olabel
+      CompactLatticeArc arc2(arc);
+      arc2.olabel = PhoneAndDurationToOlabel(phone_label,
+                                             duration, num_phones);
+      aiter.SetValue(arc2);
+     }
+  }
+}
+
+void DurationModelReplaceLabelsBackLattice(CompactLattice *clat) {
```

Comment:
You don't need this function, you just need to project on the input using  fst::Project()